example_applications/web_app/lib/web_app/prom_ex.ex

```diff
@@ -60,7 +60,11 @@ defmodule WebApp.PromEx do
       # PromEx built in plugins
       PromEx.Plugins.Application,
       PromEx.Plugins.Beam,
-      {PromEx.Plugins.Phoenix, router: WebAppWeb.Router, additional_routes: @additional_routes},
+      {
+        PromEx.Plugins.Phoenix,
+        routers: [WebAppWeb.Router, {WebAppWeb.InternalRouter, event_prefix: [:internal, :endpoint]}],
```

Comment:
Perhaps we can find a way to combine the changes you have made here, along with the changes that I have pending in this PR: https://github.com/akoutmos/prom_ex/pull/65/files#diff-68209bf72f10056cf861c9a4403e36bffa242e50c60fb547f68b4dca3c7b17dbR38

Something the along the lines of specifying an endpoint + all the routers associated with that endpoint? Since the event_prefix is also specified in the endpoint module via the telemetry plug.

I am curious if there are any introspection functions that we can use to automatically gather this information from the modules as opposed to having to duplicate this...

Reply:
I'll check if I can find a way to extract the `event_prefix` from the endpoint module.

Reply:
Sounds good :+1:

Reply:
Ok, I was able to come up with this: https://gist.github.com/emoragaf/4a5accfd1220ac0d1740608b2314bf08

But this may not work on releases if the source code is not shipped 🤔

I wasn't able to find any introspection function to extract Plug options tho